### ISSUE TITLE
Screen is dark and can't click on things after enabling crypto (5 bugs actually)

### ISSUE THREAD

**Issue by yetisyny** [Tach-Yon/Tap-Tap-Adventure#76]
The game was MOSTLY working for me, playing it at taptapadventure dot com, but I enabled crypto in the settings and then the screen turned dark, like I can still see things, everything is just darker, and also the mouse does not work anymore so I have to use the arrow keys for everything, and the only thing that I can do to interact with anything is the arrow keys.

So I can't click on any of the icons in the lower right corner such as settings or inventory or character sheet or map anymore. Those are totally inaccessible now. Also I can't change what I have equipped. And sometimes picking up items works using the arrow keys, other times it doesn't. I had a bow equipped when I turned on crypto but now with the arrow keys I have to walk right up to point blank range to shoot an enemy, defeating the purpose of using a bow, but I do it anyway, I have fought using the arrow keys with a bow equipped, also I have no armor. I can also interact with characters but it shows 2 messages in a row, the first one for a fraction of a second, then the second one. Also whenever I die I cannot click the revive button so I have to reload the webpage and then it respawns me right where I died instead of at the correct respawn point. Anyway I managed to complete the quest of killing 40 crabs despite these bugs.

I guess this is actually 5 separate bugs:

1) Screen going dark when enabling crypto and staying that way permanently even after you log out and in again.
2) Mouse not working, especially when the screen going dark bug is triggered.
3) Picking up items only sometimes working when using keyboard.
4) NPCs saying 2 lines in a row really fast so you only see their even-numbered lines.
5) If you are killed and exit/reload, you will not be at the respawn point but instead right where you were standing when you died.

So, I am reporting all 5 of those bugs, and I guess, you should keep this open until all 5 are fixed.

My username in Tap Tap Adventure is the same as my username on GitHub. So if you have a way of seeing what happened to the character “yetisyny”, you can observe this bug. Maybe.

Oh and these bugs occur in both Firefox and Chrome, latest versions. Not specific to any browser.

**Comment by Tach-Yon** [comment]
I'm planning on removing the crypto functionality. Turns out it's more of a hassle than beneficial in the long run.

**Comment by yetisyny** [comment]
Yes, most cryptocurrencies are not even remotely profitable to mine on an ordinary computer, you need an ASIC (Application-Specific Integrated Circuit) to mine them. In fact with anything based on the BitCoin algorithm, which are the vast majority of them, you will be lucky if you successfully mine anything at all, CPU mining of that died a long time ago and switched to high-end GPUs, now that is dead too and everything is done on ASICs instead of regular computers.

Only a few cryptocurrencies are profitable to mine on regular computers, such as the ones based on the ASIC-resistent CryptoNote algorithm and a couple of others. But even with these you are unlikely to make more than a few cents a day, if anything. They include Bytecoin, Monero, Ethereum, Electroneum, Zencash, and a tiny number of others, according to this article: [Cryptocurrencies you can still mine with your CPU/GPU in 2018](https://www.cointelligence.com/content/cryptocurrencies-can-still-mine-cpu-gpu-2018/).

It looks like your code mined Monero, that is one of the few that you can actually mine on a computer. Monero actually was NOT ASIC-resistant until a few days ago, the only reason it could be profitably mined on computers was that nobody had bothered to manufacture an ASIC that worked for it. Then a company started manufacturing an ASIC that would work on Monero but very soon after Monero changed its algorithm using a hard fork and now it IS ASIC-resistant: [Manufacturer Holds Cryptonight ASIC Firesale after Monero Hard Forks](https://www.ccn.com/manufacturer-holds-cryptonight-asic-firesale-after-monero-hard-forks/)

Still even though you made a good choice of which coin to mine and it IS a profitable one, a few cents a day from each computer running it is the most you can expect to make. So it would require a large user base in order for it to generate any significant amount of money. So basically, yes, more hassle than it is worth.

Better ways to make money for open-source software: having a PayPal donate button or doing crowdfunding (Bountysource, Patreon, Kickstarter, GoFundMe, etc.). Or instead of regular ads which are super-annoying, you could set up partnerships and do product placement in the game in an unobtrusive way. For example if Red Bull energy drinks or Coca-Cola or any other beverage company agreed to be a sponsor and help fund your project, you could make their beverage(s) be items in the game, that sort of thing.

All of those ways of fundraising are done by people who have YouTube channels (since YouTube drastically cut back its funding of them), other than the Bountysource one, the Bountysource one is specifically one for open-source software such as this and it integrates with GitHub, Bountysource allows users to put bounties on feature additions/bugfixes/etc. that they want and put a bounty on it and once it is resolved they pay whoever implemented the feature/fixed the bug. The open-source game OpenRA, for instance, is partly funded by Bountysource, mostly thru feature requests, integrated with GitHub.

Oh and there are also GSoC, Google Summer of Code, projects. Those usually go to bigger open-source software programs that have a lot more users, and mostly that involves getting other programmers to help you out and do internships for you and you are supposed to mentor them in programming. Very large open-source software programs with a lot of developers and users tend to do this GSoC thing. Not as common for open-source games. Actually Google Summer of Code doesn’t bring in money, instead it brings in a bright young volunteer programmer to do programming work for you temporarily.

Anyway those are a bunch of ways you can monetize free, open-source software without charging people a penny for it. Although the mining of Monero or other ASIC-resistant cryptocurrencies using JavaScript is not such a bad idea, it is just not as efficient as a bunch of other monetization methods, especially not as good as crowdfunding campaigns or just a simple PayPal donate button.

I would suggest you start out with something simple and geared towards open-source software on GitHub including games: I suggest Bountysource, let people put bounties of money on having GitHub issues resolved to their satisfaction. Then people can make feature requests (it is that more often than bugfixes) and vote with their money on what features they want implemented and if you implement what they want you get paid.

And also put a PayPal donate button on your website, that is common for open-source software too, good way to bring in donations, plus it accepts pretty much any currency. Well maybe not all the cryptocurrencies, but there are other websites people who have them can convert them to something PayPal accepts and integrate it with PayPal, not exactly sure on the details of that.

**Comment by Tach-Yon** [comment]
I’m planning on integrating my crypto into TTA later. https://weycoin.org

**Comment by Tach-Yon** [comment]
The plan is now to eventually implement a donation button and give users a donator rank and whatnot. Nothing over the top. Also, I hope you know I own a crypto, so having to go through the explanation was a bit... insulting xD

Point is, I was thinking about implementing a series of cryptos into the game, that way people are able to trade them amongst themselves

**Comment by Tach-Yon** [comment]
Number 5 is serious, I've covered the others ones by removing crypto and fixing the keyboard issue with the talking. I'll look into the item pickup when I get a chance to.
